src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunCompilerContext.cs

```diff
@@ -40,5 +40,24 @@ public override FieldLayoutAlgorithm GetLayoutAlgorithmForType(DefType type)
                 return _r2rFieldLayoutAlgorithm;
             }
         }
+
+        public override void SetSystemModule(ModuleDesc systemModule)
+        {
+            base.SetSystemModule(systemModule);
+
+            // System.Object has an EEType field in the standard AOT version used in this repo.
+            // Make sure that we always use the CoreCLR version which (currently) has no fields.
+            var objectType = systemModule.GetType("System", "Object");
+            Debug.Assert(objectType.GetField("m_pEEType") == null, "Incompatible system library. The CoreCLR System.Private.CoreLib must be used when compiling in ready-to-run mode.");
+        }
+
+        /// <summary>
+        /// CoreCLR has no Array`1 type to hang the various generic interfaces off of.
+        /// Return nothing at compile time so the runtime figures it out.
+        /// </summary>
+        protected override RuntimeInterfacesAlgorithm GetRuntimeInterfacesAlgorithmForNonPointerArrayType(ArrayType type)
+        {
+            return null;
```

Comment:
Shouldn't this be `return BaseTypeRuntimeInterfacesAlgorithm.Instance`? It should be okay to see the interfaces inherited from `System.Array`.

Reply:
Looks like it works great. Thanks!